### ISSUE TITLE
call for maintainer

### ISSUE THREAD

**Issue by suhailshergill** [suhailshergill/extensible-effects#131]
Hello all, over the next few months I might not be in a position to be an active maintainer (due to personal reasons). I would like to ensure that in the meanwhile there is minimal effect on the library. Specifically, bug-fixes and updates get pushed in a timely manner.

Could someone volunteer to be an additional maintainer? Shout out to @greydot @mattaudesse @sheyll, @power-fungus 

Thanks

**Comment by greydot** [comment]
Sure, I could try.

**Comment by suhailshergill** [comment]
Thanks @greydot !

**Comment by suhailshergill** [comment]
@greydot I've invited you to have push access to the repo. Would you like to test that with a commit to add yourself to the maintainer list in the cabal file? Once we've confirmed that github permissions are working ok, we can then sort the hackage permissions.

**Comment by sheyll** [comment]
I am also available :)

**Comment by suhailshergill** [comment]
@sheyll thanks! I'm not very knowledgeable of github's options. Would it make things easier to transition to an organization/team setup? Do you have thoughts?

**Comment by sheyll** [comment]
I have just read in the github help about the two obvious options:

1. inviting collaborators to a project
2. managing access via "Organizations".

The former option is a quick and easy way to invite someone to a single repository.

The later would allow adding repositories for a group of maintainers without re-inviting every single maintainer for every single repository. 

"Organizations" seem to reduce the amount of work to manage permissions: [Repository permission levels for an organization](https://help.github.com/articles/repository-permission-levels-for-an-organization/)

**Comment by greydot** [comment]
> @greydot I've invited you to have push access to the repo. Would you like to test that with a commit to add yourself to the maintainer list in the cabal file? Once we've confirmed that github permissions are working ok, we can then sort the hackage permissions.

I wasn't able to push to this repo.

```
> git push       
ERROR: Permission to suhailshergill/extensible-effects.git denied to greydot.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

UPD: disregard this, all sorted out.

**Comment by suhailshergill** [comment]
> it makes sense to establish an organization for extensible-effects, if this library will ever be split into multiple strongly related, yet independently released, source repositories.

that is an interesting point and one i have considered in the past. specifically, the polymorphic variants (i.e., `OpenUnion`) could potentially be a standalone library. but so far there hasn't been a compelling need to split them up.

if i'm understanding the organization roles (thanks for sharing the link btw), another feature that organizations seem to provide is that of an `admin` role. i believe "collaborators" correspond to `write` access. and so collaborators wouldn't have the ability to add/remove other collaborators (which may something we may want to have if i'm going to be away for an extended period).

**Comment by sheyll** [comment]
I believe that only owners can add or remove members, the admin role is for repo administration like, archiving and renaming.

**Comment by suhailshergill** [comment]
> I believe that only owners can add or remove members, the admin role is for repo administration like, archiving and renaming.

not quite. per the [link you shared](https://help.github.com/articles/repository-permission-levels-for-an-organization/), the below is one of the abilities that `admin` roles have:

> Add outside collaborators to a repository (see "Adding outside collaborators to repositories in your organization" for details)

similarly

> Remove outside collaborators from a repository

**Comment by sheyll** [comment]
Thanks for clarifying.

**Comment by suhailshergill** [comment]
for convenience i've decided to stick with adding collaborators (creating an organization seems a tad overkill at present). @greydot , @sheyll would you like to add your names to the maintainers list in cabal file?
btw, do you have an accounts on hackage already?

**Comment by sheyll** [comment]
First of all, thank you.

I do have a hackage account: [SvenHeyll](http://hackage.haskell.org/user/SvenHeyll)

**Comment by greydot** [comment]
@suhailshergill [sickmind](http://hackage.haskell.org/user/sickmind)

**Comment by suhailshergill** [comment]
@sheyll @greydot i've added your names to hackage maintainer list: https://hackage.haskell.org/package/extensible-effects/maintainers/
is there anything else that we need to cover that i've missed?
